specs/daito/Multi_cloud_yig_integration.md

```diff
@@ -0,0 +1,325 @@
+The purpose of this design is to add new features and analyse and facilitate the OpenSDS multi-cloud (Gelato) integration with YIG (Yet Another Index Gateway)
+
+
+One of the Key components of this is 'S3 service'. S3 service handles s3 API requests for communicating with Object Storage backends. OpenSDS multi-cloud API gateway should be compatible with any s3 compatible API service. This s3 service should provide or should be flexible to provide all the multi-cloud features. [YIG](https://github.com/journeymidnight/yig ) is Object Storage Framework which is s3 compatible. YIG brings in multiple features which fits-in into the multi-cloud requirements. 
+
+## Motivation
+
+Currently OpenSDS multi-cloud comes with a set of feature and storage backends.
+
+### Here is the list of features available in OpenSDS:
+Multi-cloud Management across public and private clouds (Data Migration, Lifecycle Management)
+Cloud Vendor Agnostic
+Policy based data mobility
+S3 based REST API
+Containerized deployment
+Metadata Management and monitoring
+Easy integration of backend storage adapters (on-prem or Cloud)
+
+### List of Object Storage backends supported:
+    * AWS
+    * Azure
+    * GCP
+    * Huawei OBS
+    * IBM COS
+    * Ceph S3
+
+### New features to be supported:
+    * Encryption of Data at rest
+    * Bucket Location
+    * Versioning
+    * CORS
+    * ACL
+    * Bucket Policy
+    * Configurable buffered copy object
+    * HEAD Bucket
+    * HEAD Object
+
+### New Object Storage Backend support: YIG
+
+#### Some of the key features of YIG:
+YIG can avoid data movement and I/O drop down caused by scaling up of Ceph cluster 
+With scale-out Ceph-cluster and ceph-libradosstripper APIs, YIG solves some of the key performance issues of Ceph
+These features make YIG Ceph Pool an important on-premise Object Storage backend.
+
+### Why YIG:
+YIG brings in some of the features which can complement and enhance OpenSDS multi-cloud management platform. This also brings in opportunity for OpenSDS to be THE Platform for all multi-cloud management. YIG has some features which OpenSDS multi-cloud doesn't have. So YIG can help adding those features into OpenSDS multi-cloud.
+
+ 
+How YIG can help in enhancing multi-cloud capabilities of Gelato:
+YIG already supports following features for the YIG Ceph pool:
+
+    * Bucket and Object ACL
+    * Bucket policy
+    * Copying Objects
+    * Server-side encryption
+    * Versioning
+    * Post object
+
```

Comment:
It is still not clear how YIG can help object store backends other than Ceph.  Please elaborate.  Can these features on YIG Ceph pool be expanded to apply on AWS S3, Azure Blob, etc.?